scripts/publish_crates-io.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+
+import argparse
+import subprocess
+import sys
+from typing import List
+import json
+
+# Using urllib instead of requests to _only_ require
+# Python 3 stdlib to work
+import urllib.request
+
+
+ALL_CRATES_IN_ORDER = ["fiberplane-pdk-macros", "fiberplane-pdk"]
+ALL_CRATES = "all"
+CRATES_IO = "crates-io"
+
+
+def index_url_path(crate: str) -> str:
+    if len(crate) == 1:
+        return f"1/{crate}"
+    elif len(crate) == 2:
+        return f"2/{crate}"
+    elif len(crate) == 3:
+        return f"3/{crate[0]}/{crate}"
+    else:
+        return f"{crate[0:2]}/{crate[2:4]}/{crate}"
+
+
+def crates_io_published_versions(crate: str) -> List[str]:
+    """
+    Fetches the current list of all published versions of a crate on crates-io.
+
+    Notably, this _includes_ the yanked versions
+    """
+    index_url = f"https://index.crates.io/{index_url_path(crate)}"
+    print(f"Requesting {index_url}")
+    request = urllib.request.Request(index_url)
```

Comment:
crates.io does not like default user agents (as gets used here) so we should be more specific with smth like `fiberplane release worker (https://github.com/fiberplane)`

Reply:
Ok!